Add contribution style guide (#3684)
ethereum/ethereum-org-website: src/content/contributing/style-guide/index.md

---
title: Style Guide
description: Style Guide for ethereum.org
lang: en
sidebar: true
---

# Ethereum.org Style Guide {#style-guide}

> Content on [Etheruem.org](http://etheruem.org) is crowdsourced and primarily written by our incredible contributors. This style guide aims to standardize certain aspects of contributing content to make the contribution process smoother.

Reviewer: ```suggestion
Content on [Etheruem.org](/) is crowdsourced and primarily written by our incredible contributors. This style guide aims to standardize certain aspects of contributing content to make the contribution process smoother.
```

## Who can submit content to Ethereum.org {#who-can-help}

> Anyone! [Ethereum.org](http://ethereum.org) is entirely open source, and many of its best pages are submitted by curious learners who expanded their notes into documentation pages now living on the site!

Reviewer: Would either remove the redundant linking to the homepage and just make this text, or if we're going to keep the links, they should just link to `/` so it's not treated as an external link.

```suggestion
Anyone! Ethereum.org is entirely open source, and many of its best pages are submitted by curious learners who expanded their notes into documentation pages now living on the site!
```
Same thing with the others below

## Audience {#audience}

[Ethereum.org](http://ethereum.org) content caters to a large and varied audience. Loosely we can categorize them as:

#### Individuals

App users, investors, enthusiasts, people who are "new to Ethereum"

#### Developers

Developers or others who want technical information about Ethereum.

#### Enterprises

People, businesses, and other organizations who want to understand Ethereum's value in an enterprise setting

> Because [Ethereum.org](http://ethereum.org) caters to a wide variety of users, all content on the site should make an effort to explain any technologies and concepts as simply as possible.

Reviewer: Would keep ethereum.org lowercased if not starting a sentence:
```suggestion
Because ethereum.org caters to a wide variety of users, all content on the site should make an effort to explain any technologies and concepts as simply as possible.
```
or 
```suggestion
Because [ethereum.org](/) caters to a wide variety of users, all content on the site should make an effort to explain any technologies and concepts as simply as possible.
```

## Objectivity {#objectivity}

[Ethereum.org](http://ethereum.org) documentation aims to maintain a credibly neutral source of truth to inform readers about Ethereum and its ecosystem. Some examples of things that we don't want in the content on Ethereum.org:

**Grand, unverifiable claims about Ethereum or adjacent technologies**

> e.g. _ "Ethereum will take over the world because..."_

**Hostile or confrontational language aimed at any organization or person**

> e.g. _ "Company X is bad because they are centralized!"_

**Politically charged rhetoric**

> > e.g. _ "This political party are better for decentralization because..."_

Reviewer: ```suggestion
> e.g. _ "This political party is better for decentralization because..."_
```

Author: @wackerow Haha that seems really weird to me. In the UK we use normally use 'are' for collective nouns. Another nuance to look out for.

## Tone {#tone}

The tone of the content on the site should be welcoming, friendly and authoritative. Where possible, jargon should be minimized and simple language used instead.

## Best Practices {#best-practices}

### Use American English {#american-english}

For words that have multiple spellings, use American English over British English.

**For example:**

- `decentralized` over `decentralised`
- `color` over `colour`

> - `analyze` over `analyse`

Reviewer: Is it official now? 😆  We've got some clean up to do

Author: Gotta pick one! I'd be more than happy to default to her Majesty's English on this occasion if we can agree on it 😂

> ### Acroynyms {#acroynyms}

Reviewer: ```suggestion
### Acronyms {#acronyms}
```

When introducing an unfamiliar acronym, spell out the full term, and put the acronym in parentheses. Put both the full term and acronym in bold.

**For example:**

> "Ethereum, like Bitcoin, currently uses a consensus protocol called **Proof-of-work (PoW)**."

Reviewer: Should we suggest here to link terms to the glossary if possible?

### Consistency {#consistency}

> Many of the topics covered on [Ethereum.org](http://ethereum.org) are technically complex. To reduce confusion to the reader, terms should be used consistently. For example, don't cycle back-and-forth between proof-of-work and PoW at random.

Reviewer: This goes to a point I made on #3653... should we encourage that the abbreviation be used after the term has been introduced on any given page?

Author: Commented in #3653 but I lean slightly towards using the full term

### Mainnet {#mainnet}

When referring to the Ethereum Mainnet (i.e. not referring to a testnet) use the proper noun. Proper nouns help avoid confusion and build greater understanding.

**Correct usage:**

- Mainnet
- Ethereum Mainnet

**Incorrect usage:**

- main net
- mainnet
- Main net
- Ethereum mainnet

> ### Proof-of-work {#proof-of-work}

Author: Combined them

Proof-of-work should be capitalized at the beginning of a sentence or when the term is introduced for the first time on a page, in any other instance, all letters should be lower case. In either case, proof-of-work should be hyphenated between each word.

**Correct usage:**

- Proof-of-work
- proof-of-work

**Incorrect usage:**

- Proof-of-Work
- Proof of work
- proof of work

### Proof-of-stake {#proof-of-stake}

The same rules we apply to proof-of-stake are applicable to proof-of-stake.

**Correct usage:**

- Proof-of-stake
- proof-of-stake

**Incorrect usage:**

- Proof-of-Stake
- Proof of stake

> - proof of stake

Reviewer: Again, just want to double-check this is accurate and proper. @ryancreatescopy You made comments about this in the past, do you recall where you found the proper english rules for these? (also commented about this in #3653)

Author: See Ryan's response in discord for context. After his input, I'm fine with keeping this as is if no one opposes it. 

But also happy to change it. My concern is that it becomes wordy and hard to follow.

### Use Active Voice {#use-active-voice}

Sentences using active voice are more concise and efficient, making your writing more engaging and easier to comprehend.

**Active voice sentence:** an actor acts on a target

> _ "The man bought a car."_

**Passive voice sentence:** a target acts on an actor

> _ "The car was bought by a man."_

[Read more on active voice](https://www.grammarly.com/blog/active-vs-passive-voice/)

> _This isn't an easy one, especially for non-native English speakers. If you aren't sure, don't worry. We'll help with this one._

Reviewer: Perhaps broaden the offer for us to help?
```suggestion
_This isn't an easy one, especially for non-native English speakers. If you aren't sure, don't worry. We'll help with any of these._
```

Author: Good idea 👍

### Linking to internal pages {#internal-links}

When linking to another page on Ethereum.org, use the relative path over the absolute path. This maintains consistency across different language versions of the site and ensures links work correctly in local development.

```md
<--- Good --->

Read more about [smart contracts](/docs/developers/smart-contracts)

<--- Bad --->

Read more about [smart contracts](https://ethereum.org/en/docs/developers/smart-contracts)
```

### Linking to images {#images}

When adding an image to a page, the image should be downloaded and placed in the same folder as the markdown file. You can reference the image like this:

`![alt text for image](./image.png)`

```md
<--- Good --->

![How to mint your NFT](./mintYourNFT.gif)

<--- Bad --->

![How to mint your NFT](https://cdn-images-1.medium.com/max/2000/0342fj_fsdfs.gif)
```

### Using Emojis {#images}

> Everyone loves emojis <Emoji text="🥰" size={1} /> To stanardize the appearence of all Emojis across browsers [Ethereum.org](http://ethereum.org) uses an `Emoji` component.

Reviewer: ```suggestion
Everyone loves emojis <Emoji text="🥰" size={1} /> To standardize the appearance of all Emojis across browsers [Ethereum.org](http://ethereum.org) uses an `Emoji` component.
```

Author: Wish I could chalk this up to intl but I just can't spell it seems.

> ---

Reviewer 2: We should link to this page from https://ethereum.org/en/contributing/, probably the "How to work on ethereum.org" section.

title: Style Guide
description: Style Guide for ethereum.org
lang: en
sidebar: true
---

# Ethereum.org Style Guide {#style-guide}

> Content on [Etheruem.org](/) is crowdsourced and primarily written by our incredible contributors. This style guide aims to standardize certain aspects of contributing content to make the contribution process smoother.

Reviewer: Another small convention we've been using throughout the site is not capitalizing the website name of ethereum.org (unless it starts a sentence).
```suggestion
Content on [etheruem.org](/) is crowdsourced and primarily written by our incredible contributors. This style guide aims to standardize certain aspects of contributing content to make the contribution process smoother.
```

> Content on [Etheruem.org](/) is crowdsourced and primarily written by our incredible contributors. This style guide aims to standardize certain aspects of contributing content to make the contribution process smoother.

Reviewer 2: Somewhere (perhaps the intro section) I think we should link to the design principles. In general I think this is a useful guide to explain the "tone/vibe" we're trying to achieve: https://ethereum.org/en/contributing/design-principles/

> ## Who can submit content to Ethereum.org {#who-can-help}

Reviewer: ```suggestion
## Who can submit content to ethereum.org {#who-can-help}
```

Anyone! Ethereum.org is entirely open source, and many of its best pages are submitted by curious learners who expanded their notes into documentation pages now living on the site!

## Audience {#audience}

> [Ethereum.org](http://ethereum.org) content caters to a large and varied audience. Loosely we can categorize them as:

Reviewer: ```suggestion
Ethereum.org content caters to a large and varied audience. Loosely we can categorize them as:
```

**Grand, unverifiable claims about Ethereum or adjacent technologies**

> e.g. _ "Ethereum will take over the world because..."_

**Hostile or confrontational language aimed at any organization or person**

> e.g. _ "Company X is bad because they are centralized!"_

**Politically charged rhetoric**

> e.g. _ "This political party is better for decentralization because..."_

## Tone {#tone}

The tone of the content on the site should be welcoming, friendly and authoritative. Where possible, jargon should be minimized and simple language used instead.

## Best Practices {#best-practices}

### Use American English {#american-english}

For words that have multiple spellings, use American English over British English.

**For example:**

- `decentralized` over `decentralised`
- `color` over `colour`
- `analyze` over `analyse`

### Acronyms {#acronyms}

When introducing an unfamiliar acronym, spell out the full term, and put the acronym in parentheses. Put both the full term and acronym in bold.

**For example:**

"Ethereum, like Bitcoin, currently uses a consensus protocol called **Proof-of-work (PoW)**."

### Consistency {#consistency}

> Many of the topics covered on [Ethereum.org](http://ethereum.org) are technically complex. To reduce confusion to the reader, terms should be used consistently. For example, don't cycle back-and-forth between proof-of-work and PoW at random.

Reviewer: ```suggestion
Many of the topics covered on ethereum.org are technically complex. To reduce confusion to the reader, terms should be used consistently. For example, don't cycle back-and-forth between proof-of-work and PoW at random.
```

### Ethereum {#ethereum}

Ethereum is a proper noun and should always be capitalized.

- `Ethereum` not `ethereum`

### Ether {#ether}

Ether is a common noun and should not be capitalized unless at the beginning of a sentence. ETH, on the other hand, is a currency abbreviation and should always be capitalized.

- `ether` not `Ether`
- `ETH` not `eth or Eth`

### Mainnet {#mainnet}

When referring to the Ethereum Mainnet (i.e. not referring to a testnet) use the proper noun. Proper nouns help avoid confusion and build greater understanding.

**Correct usage:**

- Mainnet
- Ethereum Mainnet

**Incorrect usage:**

- main net
- mainnet
- Main net
- Ethereum mainnet

### Proof-of-work / Proof-of-stake {#proof-of}

> Proof-of-work should be capitalized at the beginning of a sentence or when the term is introduced for the first time on a page, in any other instance, all letters should be lower case. In either case, proof-of-work should be hyphenated between each word.

Reviewer: > or when the term is introduced for the first time on a page

Should we be capitalizing this mid-sentence ever? I don't think capitalizing terms that are new to a page is a convention we've done elsewhere.

Reviewer: Also, perhaps we should just broaden this to proof-of-whatever.... show the PoW examples, but note that would similarly apply to proof-of-stake, proof-of-humanity, proof-of-individuality, proof-of-authority, etc.

Reviewer 2: Agreed, I don't think capitalizing mid-sentence is necessary.

```md
<!-- Good -->

Read more about [smart contracts](/docs/developers/smart-contracts)

<!-- Bad -->

Read more about [smart contracts](/en/docs/developers/smart-contracts)
Read more about [smart contracts](https://ethereum.org/en/docs/developers/smart-contracts)
```

### Linking to images {#images}

When adding an image to a page, the image should be downloaded and placed in the same folder as the markdown file. You can reference the image like this:

`![alt text for image](./image.png)`

```md
<!-- Good -->

![How to mint your NFT](./mintYourNFT.gif)

<!-- Bad -->

![How to mint your NFT](https://cdn-images-1.medium.com/max/2000/0342fj_fsdfs.gif)
```

### Using Emojis {#images}

Everyone loves emojis <Emoji text="🥰" size={1} /> To standardize the appearance of all Emojis across browsers ethereum.org uses an `Emoji` component.

```md
<--- Good --->

The London Upgrade is live <Emoji text=":heart:" size={1} />

The London Upgrade is live <Emoji text="❤️" size={1} />

<--- Bad --->

The London Upgrade is live ❤️
```

## Anything else? {#anything-else}

> Like everything on Ethereum.org, the style guide is an open-source work-in-progress with room for improvement. If there is anything you think should be added to improve this document please [edit it on GitHub](https://github.com/ethereum/ethereum-org-website/blob/dev/src/content/contributing/style-guide/index.md).

Reviewer: ```suggestion
Like everything on ethereum.org, the style guide is an open-source work-in-progress with room for improvement. If there is anything you think should be added to improve this document please [edit it on GitHub](https://github.com/ethereum/ethereum-org-website/blob/dev/src/content/contributing/style-guide/index.md).
```
